lib/Echidna/Output/Source.hs

```diff
@@ -0,0 +1,122 @@
+filterLines :: [Maybe (FilePathText, Int, TxResult)] -> [(FilePathText, Text)] -> [(FilePathText, Text)]
+filterLines []                  ls  = ls
+filterLines (Nothing      : ns) ls  = filterLines ns ls
+filterLines (Just (f,n,r) : ns) ls  = filterLines ns (markLine n r f ls)
+
+-- | Mark one particular line, from a list of lines, keeping the order of them 
+markLine :: Int -> TxResult -> FilePathText -> [(FilePathText, Text)] -> [(FilePathText, Text)] 
+markLine n r cf ls = case splitAt (n-1) ls of
+                        (xs, (f,y):ys) | f == cf -> xs ++ [(cf, markStringLine r $ unpack y)] ++ ys
+                        _                        -> map (\(f,y) -> (f, checkMarkers $ unpack y)) ls
+
+-- | Header preppend to each line
+markerHeader :: String
+markerHeader = "    |"
+
+-- | Add space for markers if necessary
+checkMarkers  :: String -> Text
+checkMarkers l@(_:_:_:_:'|':_) = pack l
+checkMarkers l                 = pack $ markerHeader ++ l
+
+-- | Add a proper marker to a line, and convert it to Text
+markStringLine  :: TxResult -> String -> Text
+markStringLine r (' ': ' ': ' ': ' ': '|':l) = pack $ getMarker r : ' ' : ' ' : ' ': '|' : l
+markStringLine r (m1: ' ': ' ': ' ':  '|':l)  = pack $ case getMarker r of
+                                                        m | m1 == m -> m1 : ' ' : ' ' : ' ': '|' : l
+                                                        m           -> m1 :  m  : ' ' : ' ': '|' : l
+
+markStringLine r (m1: m2 : ' ': ' ': '|':l)  = pack $ case getMarker r of
+                                                        m | m1 == m -> m1 :  m2  : ' ' : ' ': '|' : l
+                                                        m | m2 == m -> m1 :  m2  : ' ' : ' ': '|' : l
+                                                        m           -> m1 :  m2  : m   : ' ': '|' : l
+
+
+markStringLine r (m1: m2 : m3 : ' ': '|':l)  = pack $ case getMarker r of
+                                                        m | m1 == m -> m1 :  m2  : m3 : ' ': '|' : l
+                                                        m | m2 == m -> m1 :  m2  : m3 : ' ': '|' : l
+                                                        m | m3 == m -> m1 :  m2  : m3 : ' ': '|' : l
+                                                        m           -> m1 :  m2  : m3 : m  : '|' : l
+
+markStringLine _ (_: _ : _ : _ : '|':_)  = error "impossible to add another marker"
+markStringLine r l = pack $ getMarker r : ' ' : ' ' : ' ': '|' : l
+
+-- | Select the proper marker, according to the result of the transaction
+getMarker :: TxResult -> Char
+getMarker Success       = '*'
+getMarker ErrorRevert   = 'r' 
+getMarker ErrorOutOfGas = 'o'
+getMarker _             = 'e'
+
+-- | Given a source cache, a coverage map, a contract returns a list of covered lines
+srcMapCov :: SourceCache -> CoverageMap -> SolcContract -> [Maybe (FilePathText, Int, TxResult)]
+srcMapCov sc s c = nub $  -- Deduplicate results 
```

Comment:
`. . $` and align `.` to `=` or indent extra 2 spaces

Reply:
I think that's fixed.